### DISCUSSION TITLE
Move Demo & HelloWorld modules to Samples repo

### DISCUSSION THREAD

**Discussion by hishamco** [OrchardCMS/OrchardCore#11171] [General]
IMHO we should move `OrchardCore.Demo` and `OrchardCore.Mvc.HelloWorld` into samples repo, they are just a show case for some OC capabilities. I don't think they are in a proper place

**Comment by Skrypt** [comment]
That would make sense.

**Comment by hishamco** [comment]
I will make a PR to remove them from this repo, @agriffard could you please give me a permission on a Samples repo, so I can push a PR

**Comment by Skrypt** [comment]
I think that if we move these modules; we will need to add some documentation about these modules in the Sample repo at least. Because, else this Sample repo seems to be about showing Module functionalities.

So for example OrchardCore.Demo should be then renamed to Module3 with some explanation about what the module is about.

**Reply by Skrypt** [reply]
And I think that OrchardCore.Mvc.HelloWorld is totally useless at this point. Maybe it should be removed.

**Comment by sebastienros** [comment]
These are not samples per se. These are sandboxes and allow us to put random pieces of code that we can test or refer to internally. Some of its pieces should be also available as docs, but it doesn't mean they should be removed. Another advantage is that it's allowing us to catch breaking changes, compilations errors, ...
There is already an MVC sample in Samples.

**Reply by Piedone** [reply]
I'd keep them for these reasons, as well as to be able to do automatic renames and other refactoring. I really don't see the point in separating them (they're for Orchard contributors thus should be part of the solution).

**Comment by Skrypt** [comment]
At the same time there is this "dotnet new" module : 

[Template that allows us to test quickly an Orchard Core MVC module](https://github.com/OrchardCMS/OrchardCore/tree/main/src/Templates/OrchardCore.ProjectTemplates/content/OrchardCore.Templates.Mvc.Module)

And every Orchard Core module is also going to catch these breaking changes, compilation errors.

As for the OrchardCore.Demo module it contains samples that are generally not going to be used by anyone as is.

**Comment by hishamco** [comment]
> These are not samples per se

Also they are not fit as a modules as well ;)
